### DISCUSSION TITLE
presentation time and block instruction

### DISCUSSION THREAD

**Discussion by dylann916** [BioMotionLab/TUX#67] [Q&A] [answered]
Hello, first of all thank you for the development of your plugin, it greatly simplifies the code needed for task development. 

I am currently trying to design a very simple task in which a sequence of 2 stimuli appear one after the other. Very simply I would like to make a first stimulus (Stim1) appear for 2s and then a second stimulus (Stim2) for 200ms. As I'm a beginner I have a little trouble to do this. I think I should use the "WaitForSeconds" function but I'm not sure. I show you the beginning of my current script which works for the moment but displays both stimuli at the same time until the subject responds.

![image](https://user-images.githubusercontent.com/118400259/219062352-18500459-5df2-4411-87f9-1c70460a30ba.png)
![image](https://user-images.githubusercontent.com/118400259/219062450-a73f643c-a660-4471-8512-4bfe3901781d.png)

I also have another request (forgive me). My experiment has two different types of blocks "Easy" and "Hard" corresponding to the difficulty level of the task. I am looking for a way to display the corresponding instructions at the beginning of each block. For example, at the beginning of the "Easy" block, I would display to the subject "In this task you will have to say if the color of the two stimuli is the same or different". So for the moment I have created a block variable and two text objects corresponding to the instructions. Do you know how I can do it?

Thank you in advance, and again sorry to ask you all this.

**Comment by AdamBebko** [comment, marked as answer]
You need to use the coroutine functions in order to add in a `yield return new WaitForSeconds(1.0f)`.

As for putting something at the start of the block, You could try creating a custom block script, or using the Index properties to help with that?

It's been a while, but I also think there's a TrialInBlock property in custom trial scripts. You can take a look at the base class to check.
